### PR TITLE
Provide mode to check build durations

### PR DESCRIPTION
**Pull request by CoRfr**
This adds a mode so that check-jenkins-build-time can check the build
duration instead of when a build last occured.

The goal is to monitor the performance of some jobs.

Signed-off-by: Bertrand Roussel <broussel@sierrawireless.com>

## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out on [Our CHANGELOG Guidelines](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

**Comment by CoRfr** [comment]
Sample result:

Passed:
```
$ check-jenkins-build-time.rb -u 'http://jenkins-cnt:8080/' --jobs 'Legato-Review=1h' --check-build-duration
JenkinsBuildTime OK: 1 job(s) had successful builds within allowed times
```
Failed:
```
$ check-jenkins-build-time.rb -u 'http://jenkins-cnt:8080/' --jobs 'Legato-Review=30m' --check-build-duration
JenkinsBuildTime CRITICAL: 1 failure(s): Legato-Review: last built at 2018-05-10 01:47:18 +0000 (33 mins 41 secs) exceeded max duration = 30m
```

File: bin/check-jenkins-build-time.rb [hunk 1]
```diff
@@ -104,9 +121,21 @@ def last_successful_build_number(job_name)
...
+  end
+
   def build_time(job_name, build_number)
     # Jenkins expresses timestamps in epoch millis
-    Time.at(jenkins.job.get_build_details(job_name, build_number)['timestamp'] / 1000)
+    Time.at(build_details(job_name, build_number)['timestamp'] / 1000)
+  end
+
+  def build_duration(job_name, build_number)
+    # Jenkins expresses timestamps in epoch millis, convert it to seconds
```
Comment (majormoses) [inline bin/check-jenkins-build-time.rb:137]: :+1:

**Review by majormoses — APPROVED** [review]
LGTM

I had to read it multiple times because I could not figure out where the duration was until I opened the whole file and saw that the jobs includes a time window and you are just reusing those values.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,8 @@ This CHANGELOG follows the format listed at [Our CHANGELOG Guidelines](https://g
 Which is based on [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+- check-jenkins-build-time.rb: add `--check-build-duration` mode to check duration (@CoRfr)
```
Comment (majormoses) [inline CHANGELOG.md:8]: These should go under an `### Added` sub header per our [changelog guidelines](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-jenkins/versions/1.7.0

**Pull request merged**